librz/core/cfile.c

```diff
@@ -120,25 +126,35 @@ static bool __rebase_xrefs(void *user, const ut64 k, const void *v) {
 	return true;
 }
 
-static void __rebase_everything(RzCore *core, RzList *old_sections, ut64 old_base) {
+
+/**
+ * \brief rebase all flags, binary, information, breakpoints, and analysis.
+ * \param core Rizin core.
+ * \param sections_backup Backup of original file sections.
+ * \param infer_new_baddr_shift Should the object's current baddr_shift be used to compute a new base address.
+ * \return void
+ */
+RZ_API void rz_core_rebase_everything(RZ_BORROW RzCore *core, RZ_BORROW RzList *sections_backup, bool infer_new_baddr_shift, ut64 old_baddr_shift, ut64 new_baddr_shift) {
```

Comment:
It's also a good idea to use the `RZ_NONNULL` annotation for the arguments which you don't expect to be `NULL`. Like `sections_backup` in this case.

Also add the `rz_return_val_if_fail` check here as well (you should check both, `core` and `sections_backup`).

Side note: Usually, we don't use the `RZ_NONNULL` annotation with `RzCore` because of brevity and to reduce unnecessary clutter. It is implicit that any `RzCore` pointer being passed should not be `NULL`

Reply:
I also suggest removing `RZ_BORROW` from the `RzCore` as well.